src/libraries/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEnumerator.cs

```diff
@@ -38,6 +40,8 @@ internal FileSystemEnumerator(string directory, bool isNormalized, EnumerationOp
             _rootDirectory = Path.TrimEndingDirectorySeparator(path);
             _options = options ?? EnumerationOptions.Default;
 
+            _remainingDepth = _options.MaxRecursionDepth < 0 ? EnumerationOptions.Default.MaxRecursionDepth : _options.MaxRecursionDepth;
+
```

Comment:
> @iSazonov I don't expect we will ever change the default value of `MaxRecursionDepth`, mainly because it's type (`int`) would never be changed.

I believe it makes the code more readable if we use a sort of named constants. 
Also I guess/hope the runtime will optimize the case.

Waiting @adamsitnik ...

Reply:
+1 to use a named constant (perhaps called `DefaultMaxRecursionDepth`) inside `EnumerationOptions` and use it where its necessary.

Reply:
Done.